NDTensors/src/lib/GradedAxes/src/dual.jl

```diff
@@ -5,3 +5,5 @@ using NDTensors.LabelledNumbers:
 label_dual(x) = label_dual(LabelledStyle(x), x)
 label_dual(::NotLabelled, x) = x
 label_dual(::IsLabelled, x) = labelled(unlabel(x), dual(label(x)))
+
+flip(g::AbstractGradedUnitRange) = dual(gradedrange(label_dual.(blocklengths(g))))
```

Comment:
The following doesn't have to be implemented in this PR, but our discussion in https://github.com/ITensor/ITensors.jl/pull/1531#issuecomment-2358717612 made me think this and related functions could be written in a nicer way.

For example, it could be nice if we could write this function as:
```julia
flip(g::AbstractGradedUnitRange) = gradedrange(flip.(blocklengths(g)))
```
In order to make that work, we would need to have `flip` work on sectors, define the concept of dual sectors, and make it so that `gradedrange` outputs a dual unit range when constructed from dual sectors.

An implementation could be the following:
```julia
flip(l::LabelledInteger) = labelled(unlabel(l), flip(label(l)))

struct CategoryDual{NondualCategory<:AbstractCategory} <: AbstractCategory
  nondual_category::NondualCategory
end
flip(c::AbstractCategory) = CategoryDual(dual(c))
nondual(c::AbstractCategory) = c
nondual(c::CategoryDual) = c.nondual_category
dual(c::CategoryDual) = nondual(c)
```
In general, `dual(flip(x))` can be the canonical/generic way of instantiating a dual sector/space/range, and could supersede `label_dual` (which I never liked but didn't have a better idea for at the time).

Additionally, `gradedrange` would have to be able to detect that all of the labels are `CategoryDual` and in that case construct a `UnitRangeDual`/`GradedUnitRangeDual`.

Again, that's probably outside the scope of this PR, just putting it out there as an idea and we can make a comment/issue to track that we might want to do this later.

Reply:
I re-commented this here: https://github.com/ITensor/ITensors.jl/pull/1531#issuecomment-2393864138 so I'll resolve this comment thread.